packages/lit-dev-content/site/tutorials/content/svg-templates/03.md

```diff
@@ -0,0 +1,123 @@
+This section of the tutorial covers the basics of `<clipPath>` and
+`<g>` elements in SVG including how to:
+
+- Create a clip path
+- Create a group of groups with `<use>`
+- Apply a clip path to a group
+
+### Learn
+
+In pattern making, a _tile_ is a polygon that can be repeated across
+a two dimensional plane. The tile in this demo will be a series of
+transformed texts.
+
+Clip paths are polygons used to restrict where elements are "painted".
+In SVG, a clip path is defined with the `<clipPath>` element.
+
+In the example below, a clip path will be created in the shape of a 
+rectangle with an `id` of `rect-clip`.
+
+```ts
+const helloClipPath = svg`
+	<clipPath>
+		<rect width="200" height="200"></rect>
+	</clipPath>
+`;
+```
+
+In order to apply a clip path to an SVG element, the `clip-path`
+property should reference a `<clipPath>` element by an `id`. The
+`<rect>` below will be clipped by the clip-path above by referencing
+`#react-clip`.
+
+```ts
+const helloTile = svg`
+	<rect
+		clip-path="url(#rect-clip)"
+		width="300"
+		height="300"
+		fill="#000">
+	</rect>
+```
+
+### Apply
+
+Create a clip path composed of a 200px square. This `<clipPath>` will
+restrict paints to create a _tile_. 
+
+```ts
+const createClipPath = () => svg`
+	<clipPath id="rect-clip">
+		<rect width="200" height="200"></rect>
+	</clipPath>
+`;
+```
+
+Then give the motif an `id` called `chars-rotated`.
+
+```ts
+const createMotif = (
+	numPrints: number,
+	offset: number = 0,
+): SVGTemplateResult => {
+
+	...
+
+	return svg`
+		<g
+			id="chars-rotated"
+			transform="translate(50, 50)">
+				${prints}
+		</g>`;
+}
+```
+
+After, move the _motif_ inside `<defs>`.
+
+```ts
+@customElement('repeat-pattern')
+export class RepeatPattern extends LitElement {
+
+	...
+
+	render() {
+		return html`
+			<svg>
+				<defs>
+					${createClipPath()}
+					${createElement(this.chars)}
+					${createMotif(
+						this.numPrints,
+						this.rotationOffset,
+					)}
+				</defs>
+			</svg>
+    	`;
+	}
+}
+```
+
+Now that the motif is available in `<defs>`, make a function called
+`createTile` that returns a group of rotated texts via `<use>`.
```

Comment:
You never say to use it

Reply:
changed the `#chars-rotated` id to `#motif` to make it more clear